x-pack/plugin/fleet/src/internalClusterTest/java/fleet/action/GetGlobalCheckpointsActionIT.java

```diff
@@ -0,0 +1,236 @@
+            .toArray();
+        assertArrayEquals(fromStats, response2.globalCheckpoints());
+    }
+
+    public void testPollGlobalCheckpointAdvancement() throws Exception {
+        String indexName = "test_index";
+        client().admin()
+            .indices()
+            .prepareCreate(indexName)
+            .setSettings(
+                Settings.builder()
+                    .put(IndexSettings.INDEX_TRANSLOG_DURABILITY_SETTING.getKey(), Translog.Durability.REQUEST)
+                    .put("index.number_of_shards", 1)
+                    .put("index.number_of_replicas", 1)
+            )
+            .get();
+
+        final GetGlobalCheckpointsAction.Request request = new GetGlobalCheckpointsAction.Request(
+            indexName,
+            false,
+            EMPTY_ARRAY,
+            TEN_SECONDS
+        );
+        final GetGlobalCheckpointsAction.Response response = client().execute(GetGlobalCheckpointsAction.INSTANCE, request).get();
+        assertEquals(-1, response.globalCheckpoints()[0]);
+
+        final int totalDocuments = between(25, 50);
+        new Thread(() -> {
+            for (int i = 0; i < totalDocuments; ++i) {
+                client().prepareIndex(indexName).setId(Integer.toString(i)).setSource("{}", XContentType.JSON).execute();
+            }
+
+        }).start();
+
+        final GetGlobalCheckpointsAction.Request request2 = new GetGlobalCheckpointsAction.Request(
+            indexName,
+            true,
+            new long[] { totalDocuments - 2 },
+            TimeValue.timeValueSeconds(30)
+        );
+        final GetGlobalCheckpointsAction.Response response2 = client().execute(GetGlobalCheckpointsAction.INSTANCE, request2).get();
+        assertEquals(totalDocuments - 1, response2.globalCheckpoints()[0]);
+    }
+
+    public void testPollGlobalCheckpointAdvancementTimeout() throws Exception {
+        String indexName = "test_index";
+        client().admin()
+            .indices()
+            .prepareCreate(indexName)
+            .setSettings(
+                Settings.builder()
+                    .put(IndexSettings.INDEX_TRANSLOG_DURABILITY_SETTING.getKey(), Translog.Durability.REQUEST)
+                    .put("index.number_of_shards", 1)
+                    .put("index.number_of_replicas", 0)
+            )
+            .get();
+
+        final int totalDocuments = 30;
+        for (int i = 0; i < totalDocuments; ++i) {
+            client().prepareIndex(indexName).setId(Integer.toString(i)).setSource("{}", XContentType.JSON).execute();
+        }
+
+        final GetGlobalCheckpointsAction.Request request = new GetGlobalCheckpointsAction.Request(
+            indexName,
+            true,
+            new long[] { 29 },
+            TimeValue.timeValueMillis(between(0, 100))
+        );
+        GetGlobalCheckpointsAction.Response response = client().execute(GetGlobalCheckpointsAction.INSTANCE, request).actionGet();
+        assertTrue(response.timedOut());
```

Comment:
Can we also assert that:

1. The correct sequence numbers are returned.
2. It took less than 30 seconds to timeout.